common/common.cpp

```diff
@@ -647,9 +681,15 @@ void gpt_print_usage(int /*argc*/, char ** argv, const gpt_params & params) {
     printf("  --cfg-negative-prompt-file FNAME\n");
     printf("                        negative prompt file to use for guidance. (default: empty)\n");
     printf("  --cfg-scale N         strength of guidance (default: %f, 1.0 = disable)\n", params.cfg_scale);
-    printf("  --rope-scale N        RoPE context linear scaling factor, inverse of --rope-freq-scale\n");
+    printf("  --rope-scaling {none,linear,yarn}\n");
+    printf("                        RoPE frequency scaling method, defaults to linear unless specified by the model\n");
+    printf("  --rope-scale N        RoPE context scaling factor, inverse of --rope-freq-scale\n");
     printf("  --rope-freq-base N    RoPE base frequency, used by NTK-aware scaling (default: loaded from model)\n");
-    printf("  --rope-freq-scale N   RoPE frequency linear scaling factor (default: loaded from model)\n");
+    printf("  --rope-freq-scale N   RoPE frequency scaling factor (default: loaded from model)\n");
+    printf("  --yarn-ext-factor N   YaRN extrapolation mix factor (default: %.1f)\n", params.yarn_ext_factor);
+    printf("  --yarn-attn-factor N  YaRN magnitude scaling factor (default: %.1f)\n", params.yarn_attn_factor);
+    printf("  --yarn-beta-fast N    YaRN low correction dim (default: %.1f)\n", params.yarn_beta_fast);
+    printf("  --yarn-beta-slow N    YaRN high correction dim (default: %.1f)\n", params.yarn_beta_slow);
```

Comment:
depending on how close the the paper the names of the parameters are, we might want to add a readme with more details.

Reply:
I don't believe the extrapolation mix factor is documented in the paper because it describes an implementation that assumes it is always `1.0`. The slow and fast beta are known as $\alpha$ and $\beta$ in the paper, respectively. attn_factor is known as $\sqrt t$ in the paper, and I've just noticed that the python implementation never sets it but the paper says it should be set to about $0.1 \ln(s) + 1$, where $s$ is the scale factor.